core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ccm/UpgradeCcmOrchestratorService.java

```diff
@@ -0,0 +1,75 @@
+package com.sequenceiq.cloudbreak.service.upgrade.ccm;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.orchestration.Node;
+import com.sequenceiq.cloudbreak.core.bootstrap.service.ClusterDeletionBasedExitCriteriaModel;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorException;
+import com.sequenceiq.cloudbreak.orchestrator.host.HostOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.host.OrchestratorStateParams;
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+
+@Service
+public class UpgradeCcmOrchestratorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpgradeCcmOrchestratorService.class);
+
+    private static final String NGINX_STATE = "nginx";
+
+    private static final String DISABLE_MINA_STATE = "upgradeccm/disable-ccmv1";
+
+    private static final String DISABLE_INVERTING_PROXY_AGENT_STATE = "upgradeccm/disable-ccmv2";
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private StackUtil stackUtil;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
```

Comment:
I know this is not in the environment service - where we usually do injection in the following way -, but can we replace the field injection with constructor injection, since the first one is not a preferred way of injection even from the guys of Spring.
Having a new class with such a few dependencies is a pretty good chance to improve the overall code quality :)

Reply:
I prefer @Inject as if I need to add more services then the large number of constructor parameters is also an antipattern.

Reply:
Yes but having tons of constructor parameters is a good indicator for that the class does much more than it should